### PR TITLE
feat: upgrading Mender Server: upgrading to 3.5

### PR DESCRIPTION
**Pull request by merlin-northern**
feat: upgrading Mender Server: upgrading to 3.5

Changelog: Title
Ticket: MEN-6219
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@TheMeaningfulEngineer could you give me review only on the 3.5 chapter here? (the rest will disappear from there when we merge https://github.com/mendersoftware/mender-docs/pull/2036)

File: 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+Lack in the above can result in the following log line from opensearch:
+
+```shell
+[1]: max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
+ERROR: OpenSearch did not exit normally - check the logs at /usr/share/opensearch/logs/opensearch-cluster.log
+```
+
+If that happens, free some space and restart your setup.
+
+### Database backup
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md:49]: We cover this already in the general section.
Thought the idea was to not have to repeat it every time.

File: 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+Be sure to have a fresh and working backup of your database. The perfect solution
+is to have a testing environment setup where you can restore the backup, and see that
+everything is in order, and try the upgrade.
+
+### Access to all the images
+
+Remember that during the upgrade process your orchestrator will try to download images.
+Be sure that you have access to them.
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md:58]: What does this mean?
The images of the new container components?

**Review by TheMeaningfulEngineer — CHANGES_REQUESTED** [review]
I think there are chapters in both 3.4 and 3.5 that can be moved to 'generic'

Same structure comment as for 3.4: 
https://github.com/mendersoftware/mender-docs/pull/2040#discussion_r1122873930

File: 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md]: yes, kind of obvious. I removed it.

File: 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md]: removed. (left orver)

File: 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+taxonomy:
+category: docs
+label: tutorial
+---
+
+This is a tutorial for upgrading the Mender Server to version 3.5 in production environments.
+With the 3.5 we introduced a new element at the heart of ths system:
+the so-called _reporting_ service, which came with [OpenSearch](https://opensearch.org/).
+This brings an additional upgrade complication, and needs some attention.
+
```
Comment (TheMeaningfulEngineer) [inline 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md:12]:
```suggestion

Besides the version specific guidelines below, please make sure you're following the [suggested upgrade best practices](../docs.md) to reduce the risk of unpredictable downtime and data loss.
```

Comment (merlin-northern) [reply 07.Server-installation/02.Upgrading-from-previous-versions/100.Upgrading-to-3.5/docs.md:12]: sure.

**Review by TheMeaningfulEngineer — APPROVED** [review]

**Pull request merged**